leetcodePython/PrefixSum/prefixsum_560.py

```diff
@@ -0,0 +1,19 @@
+from typing import List
+from collections import Counter
+
+class Solution:
+    def subarraySum(self, nums: List[int], k: int) -> int:
+        cumul_sum = 0
+        ans = 0
+        cumul_sum_count = Counter()
+        for num in nums:
+            cumul_sum += num
+            if cumul_sum == k:
+               ans += 1
+            diff = cumul_sum - k
+
+            # k = curr_sum - diff
+            # the count here shows how many cumulate sub array we can subtract to form a valid sub array
+            ans += cumul_sum_count[diff]
```

Comment:
## Hint

1. We are calculating subarray (subset) problem. We need to a way to count.
2. `Prefix sum` is aways consecutive and cumulative. It means each index we form a subarray (subset) from beginning.